src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1703,7 +1707,7 @@ abstract class RefChecks extends Transform {
               assert(sym != NoSymbol, "transformCaseApply: name = " + name.debugString + " tree = " + tree + " / " + tree.getClass) //debug
               enterReference(tree.pos, sym)
             }
-            tree
+            eliminatedRightAssocValDefs.getOrElse(tree.symbol, tree)
```

Comment:
Hm, I assumed it would be safe here because I'm only moving it within the same block. But maybe that's no longer true in all cases after uncurry?